src/extensions/magic_mount/chunk.rs

```diff
@@ -0,0 +1,116 @@
+                    &format!("{}:{}", key_name, chunk_id),
+                    &payload[offset..offset + chunk_size],
+                )
+                .await?;
+            chunk_paths.push(response);
+            offset += chunk_size;
+            chunk_id += 1;
+        }
+
+        // store the chunk paths in the metadata entry and update metadata
+        self.update_entry(&metadata_key.clone(), &chunk_paths.join(";").into_bytes())
+            .await
+    }
+
+    #[async_recursion]
+    pub(crate) async fn _read_entry_chunk(&self, key_name: &str) -> Result<Vec<u8>, Error> {
+        let metadata_key = format!("{}:chunk_metadata", key_name);
+        let metadata_response = self.read_entry(&metadata_key.clone()).await?;
+
+        // check if the metadata is locked
+        let payload_string = String::from_utf8(metadata_response.clone())?;
+        if payload_string == "creation-in-progress-locked" {
+            return Err("Creation in progress".into());
+        }
+
+        // read the chunks into a single vector
+        let chunks = payload_string.split(';').collect::<Vec<&str>>();
+        let mut payload = Vec::new();
+        for chunk in chunks {
+            let response = self.read_entry(&format!("{}:{}", key_name, chunk)).await?;
+            payload.push(response[0].clone());
+        }
+        Ok(payload)
+    }
+
+    #[async_recursion]
+    pub(crate) async fn _update_entry_chunk(
+        &self,
+        key_name: &str,
+        payload: &[u8],
+    ) -> Result<String, Error> {
+        let metadata_key = format!("{}:chunk_metadata", key_name);
+
+        // check if the metadata is locked
+        let response = self.read_entry(&metadata_key.clone()).await?;
+        let payload_string = String::from_utf8(response.clone())?;
+        if payload_string == "creation-in-progress-locked" {
+            return Err("Creation in progress".into());
+        }
+
+        // split payload into chunks and update the chunks
+        let mut offset = 0;
+        let mut chunk_id = 0;
+        let mut chunk_paths = Vec::new();
+        while offset < payload.len() {
+            let chunk_size = if offset + CHUNK_SIZE > payload.len() {
+                payload.len() - offset
+            } else {
+                CHUNK_SIZE
+            };
+            let response = self
+                .update_entry(
+                    &format!("{}:{}", key_name, chunk_id),
+                    &payload[offset..offset + chunk_size],
+                )
+                .await?;
+            chunk_paths.push(response);
+            offset += chunk_size;
+            chunk_id += 1;
+        }
+
+        // update metadata with new chunk paths
+        self.update_entry(&metadata_key.clone(), &chunk_paths.join(";").into_bytes())
+            .await
+    }
+
+    #[async_recursion]
+    pub(crate) async fn _delete_entry_chunk(&self, key_name: &str) -> Result<String, Error> {
+        let metadata_key = format!("{}:chunk_metadata", key_name);
```

Comment:
why not check `creation-in-progress-locked` here?

Imagine the following scenario:
1. create A, create the metadata "in-progress" (still running)
2. delete A, metadata_key is deleted
3. create A (again), create the metadata "in-progress" again
Now, who is going to finish first between (1) and (3) will result in different outcomes, where semantically, it should always be (3) who makes the last update.
Instead, we can probably just throw an error in step 2, which makes it much simpler.